Assets/Scripts/WinOrLoseScript.cs

```diff
@@ -54,88 +164,266 @@ void Start()
+
+            foreach (Player player in ties)
+            {
+                if (IsLocalPlayer(player))
+                {
+                    PlayWinOrLose(player);
+                }
+            }
+        }
+
+        //when all places are displayed
+        DisableLastPlacePodium();
+    }
+    #endregion
+
+    #endregion
+
+    #region organizing
+
+    #region Tie Logic
+
+    #region IsTied
+    bool IsTied(int index1, int index2)
+    {
+        return IsTied(players[index1], players[index2]); //overload
     }
 
-    IEnumerator Flash()
+    bool IsTied(Player player1, Player player2)
     {
-        //int incrementation = 0;
-        //foreach (Player player in players)
-        //{
-            yield return new WaitForSeconds(6);
-        //    //images[incrementation].gameObject.SetActive(true);
-        //    incrementation++;
-        //}
+        return player1.score == player2.score;
+    }
+    #endregion
+
+    private void CheckForTies()
+    {
+        winners.Add(players[0]); //add the known winner to the list
+
+        for(int i = 0; i < players.Length; i++)
+        {
+            if(i != 0)
+            {//if not first player (blocks error)
+                if(IsTied(players[i], players[i - 1]))
+                {//if player is tied with the one before it
+                    ReplacePodium(i);
+
+                    if (IsTied(players[i], winners[0]))
+                    {//if player is tied with the winner
+                        winners.Add(players[i]);
+                    }
+                }
+            }
+        }
 
-        podiums[players.Length-1].gameObject.SetActive(false);
+        PopulateLosers();
     }
+    #endregion
 
-    Player[] Order(Player[] scores)
+    private void PopulateLosers()
     {
-        for(int current = 0; current < scores.Length; current++)
+        foreach(Player player in players)
         {
-            for(int after = current + 1; after < scores.Length; after++)
+            if (!winners.Contains(player))
+            {//if not a winner
+                losers.Add(player);
+            }
+        }
+    }
+
+    private List<int> GenerateBigestLosersList()
+    {
+        if(losers.Count == 0)
+        {
+            return new List<int>(new int[] { 0, 1, 2, 3 });
+        }
+
+        int index = losers.Count - 1; //starts on player in last
+
+        //Generate list of last place players
+        List<int> podiumsToDestroy = new List<int>();
+        podiumsToDestroy.Add(index); //add known biggest loser to list
+
+        while (index - 1 != -1 && IsTied(losers[index], losers[index - 1]))
+        {//while in bounds and has the same score as last place person
+            //add to the destroy list
+            podiumsToDestroy.Add(index - 1);
+
+            //setup for next loop
+            index--;
+        }
+
+        return podiumsToDestroy;
+    }
+
+    void Order()
+    {
+        for(int current = 0; current < players.Length; current++)
```

Comment:
we already have linq in the project. You could just do  Players.OrderBy(Score)
https://www.tutorialsteacher.com/linq/linq-sorting-operators-orderby-orderbydescending

Reply:
Completed